modules/apps/announcements/announcements-web/src/main/java/com/liferay/announcements/web/internal/change/tracking/spi/display/AnnouncementsDeliveryCTDisplayRenderer.java

```diff
@@ -23,16 +38,55 @@
 public class AnnouncementsDeliveryCTDisplayRenderer
 	extends BaseCTDisplayRenderer<AnnouncementsDelivery> {
 
+	@Override
+	public String getEditURL(
+		HttpServletRequest httpServletRequest,
+		AnnouncementsDelivery announcementsDelivery) {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		if (!UserPermissionUtil.contains(
+				themeDisplay.getPermissionChecker(), themeDisplay.getUserId(),
```

Comment:
This will always return true because user always has permission to update themselves. We should check permission on the user associated to the announcementsDelivery not the current user. themeDisplay.getUserId() is the current logged in user.